clients/src/main/java/org/apache/kafka/common/config/AbstractConfig.java

```diff
@@ -355,13 +357,20 @@ public Map<String, Object> valuesWithPrefixAllOrNothing(String prefix) {
         return nonInternalConfigs;
     }
 
-    private void logAll() {
+    /**
+     * Won't do any filter in the abstract config, but can be overridden in subclasses.
+     */
+    protected Map<String, Object> clearUnsupportedConfigsForLogging(Map<String, Object> values) {
+        return new TreeMap<>(values);
+    }
+
+    private void logAll(Map<String, Object> values) {
         StringBuilder b = new StringBuilder();
         b.append(getClass().getSimpleName());
         b.append(" values: ");
         b.append(Utils.NL);
 
```

Comment:
That way we can invoke this from inside `logAll()`, keeping the logging bits closer together.

```suggestion
        Map<String, Object> valuesToLog - clearUnsupportedConfigsForLogging(values);
```